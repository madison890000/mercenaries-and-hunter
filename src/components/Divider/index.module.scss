.divider {
  margin: 10px 0 0 0;
  width: 100%;
}

.section {
  margin: 10px 0;
  border: 2px solid gray;
}
