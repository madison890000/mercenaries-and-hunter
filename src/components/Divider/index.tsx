import React from 'react';
import styles from './index.module.scss';
import capitalize from '../../utils/capitalize';

interface DividerProps {
    title?: string;
    extra?: React.ReactNode;
}

const Divider = ({ title, extra }: DividerProps) => {
    return (
        <div className={styles.divider}>
            <div className={styles.title}>
                <div>
                    <h3>{capitalize(title)}</h3>
                </div>
                <div>{extra}</div>
            </div>
            <div className={styles.section} />
        </div>
    );
};

export default Divider;
