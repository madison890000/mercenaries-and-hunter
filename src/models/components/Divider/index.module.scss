.dividerH {
  border: var(--base-gap-unit) solid var(--divider-border-color);
  width: 0px;
  height: auto;
  display: block;
  min-height: 30px;
  margin: 0 12px;
}

.divider {
  margin: var(--block-margin) 0 0 0;
  width: 100%;

  .title {
    font-size: var(--base-font-size-large);
    font-weight: bold;
    margin: 18px 0;
  }
}

.section {
  margin: 10px 0;
  border: var(--base-gap-unit) solid var(--divider-border-color);
}

.sectionDash {
  margin: 10px 0;
  border: var(--base-gap-unit) dashed var(--divider-border-color);
}
