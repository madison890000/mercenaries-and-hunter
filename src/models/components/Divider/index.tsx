import React from 'react';
import styles from './index.module.scss';
import capitalize from '../../../utils/capitalize';

interface DividerProps {
    title?: string;
    variant?: 'v' | 'dash';
}

const Divider = ({variant, title}: DividerProps) => {
    if (variant === 'v') {
        return <div className={styles.dividerH}/>;
    }
    if (variant === 'dash') {
        return <div className={styles.sectionDash}/>;
    }
    return (
        <div className={styles.divider}>
            <div>
                <div className={styles.title}>{capitalize(title)}</div>
            </div>
            <div className={styles.section}/>
        </div>
    );
};

export default Divider;
