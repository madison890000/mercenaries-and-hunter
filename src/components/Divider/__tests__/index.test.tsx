import React from 'react';
import Divider from '../index';
import renderer from 'react-test-renderer';

test('renders Divider', () => {
    const DividerDom = renderer.create(<Divider title="test Divider"/>).toJSON();
    expect(DividerDom).toMatchSnapshot();
});


test('renders Divider -> variant dash', () => {
    const DividerDom = renderer.create(<Divider variant="dash" title="test Divider"/>).toJSON();
    expect(DividerDom).toMatchSnapshot();
});

test('renders Divider -> variant v', () => {
    const DividerDom = renderer.create(<Divider variant="v" title="test Divider"/>).toJSON();
    expect(DividerDom).toMatchSnapshot();
});
