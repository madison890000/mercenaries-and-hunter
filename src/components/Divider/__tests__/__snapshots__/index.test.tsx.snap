// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Divider 1`] = `
<div
  className="divider"
>
  <div
    className="title"
  >
    <div>
      <h3>
        Test Divider
      </h3>
    </div>
    <div />
  </div>
  <div
    className="section"
  />
</div>
`;

exports[`renders Divider with extra 1`] = `
<div
  className="divider"
>
  <div
    className="title"
  >
    <div>
      <h3>
        Test Divider
      </h3>
    </div>
    <div>
      test extra
    </div>
  </div>
  <div
    className="section"
  />
</div>
`;
